src/components/date/__internal__/date-formats/index.js

```diff
@@ -34,37 +34,122 @@ const NA_FORMATS = [
   "MM dd yyyy",
 ];
 
+// The order of this array is important, when an input value matches more than one format the last one is used
+const CN_FORMATS = [
+  "yyyy M",
+  "yyyy M d",
+  "yyyy MM d",
+  "yyyy M dd",
+  "yyyy MM dd",
+  "yy M",
+  "yy MM",
+  "yy M d",
+  "yy MM d",
+  "yy M dd",
+  "yy MM dd",
+  "M",
+  "M d",
+  "MM",
+  "M dd",
+  "MM d",
+  "MM dd",
+];
+
 const SEPARATORS = ["", ".", ",", "-", "/", ":"];
 
-const generateFormats = (formatArray) =>
-  formatArray.reduce((arr, formatString) => {
+const STANDARD_FORMAT_LENGTH = 10;
+
+const generateFormats = (formatArray, separator, trailingChar) => {
+  const separators = SEPARATORS.includes(separator)
+    ? SEPARATORS
+    : [...SEPARATORS, separator];
+
+  return formatArray.reduce((arr, formatString) => {
     const array = [...arr, formatString];
     if (formatString.includes(" ")) {
-      SEPARATORS.forEach((char) =>
-        array.push(formatString.replace(/ /g, char))
-      );
+      separators.forEach((char) => {
+        if (separator === char && trailingChar) {
+          array.push(`${formatString.replace(/ /g, char)}${trailingChar}`);
+        }
+        array.push(formatString.replace(/ /g, char));
+      });
     }
     return array;
   }, []);
+};
+
+const getOutputFormatForLocale = (localeCode) => {
+  const formatMap = {
+    day: "dd",
+    month: "MM",
+    year: "yyyy",
+  };
+
+  const formatter = new Intl.DateTimeFormat(localeCode);
+  let separator;
+
+  const format = formatter
+    .formatToParts(new Date())
+    .map(({ type, value }) => {
+      if (type !== "literal") {
+        return formatMap[type];
+      }
+      if (!separator) {
```

Comment:
```suggestion
      else if (!separator) {
```

Not sure if this needs to be an `else` so `separator` is only set to the `value` corresponding to the first `type: literal` found in the array, so its guaranteed `separator` will be set to a separator character and not one associated with a day, month, year, etc.?

Reply:
The `return` statement on line 95 would prevent the `if` being reached so there's no need for the `else`